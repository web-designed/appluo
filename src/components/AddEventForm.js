import React from 'react'
import { connect } from 'react-redux'
import uuid from 'uuid'
import moment from 'moment'
import 'react-dates/initialize'
import { SingleDatePicker } from 'react-dates'
import { Link } from 'react-router-dom'

export class AddEventForm extends React.Component {

   state = {
      date: this.props.cleanedAt ? moment(this.props.cleanedAt) : moment(),
      focused: false,
      note: this.props.note ? this.props.note : '',
      place: this.props.place ? this.props.place : this.props.settings.places[0],
      error: '',
      cleaner: this.props.cleaner ? this.props.cleaner : this.props.settings.cleaners[0],
   }

   onCleanerChange = (e) => {
      const cleaner = e.target.value
      this.setState(() => ({ cleaner }))
   }

   onPlaceChange = (e) => {
      const place = e.target.value
      this.setState(() => ({ place }))
   }

   onNoteChange = (e) => {
      const note = e.target.value
      this.setState(() => ({ note }))
   }

   onDateChange = (date) => {
      this.setState(() => ({ date }))
   }

   onFocusChange = ({focused}) => {
      this.setState(() => ({focused}))
   }

   handleSubmit = (e) => {

      e.preventDefault()
      
      const event = {
         cleanedAt: this.state.date.valueOf(),
         createdAt: moment().valueOf(),
         cleaner: this.state.cleaner,
         place: this.state.place,
         note: this.state.note,
      }

      this.props.handleSubmit(event)
   }

   render(){

      console.log(this.props)

      return(
         <div>
            <form onSubmit={this.handleSubmit}>
               <div class="input-group mb-2">
                  <div class="input-group-prepend">
                     <label class="input-group-text" for="cleaner">Name</label>
                  </div>
                  <select 
                     class="custom-select" 
                     id="cleaner"
                     name="cleaner" 
                     onChange={this.onCleanerChange}
                     value={this.state.cleaner.toLowerCase()}
                  >
                     {
                        this.props.settings.cleaners.map( cleaner => {
                           return <option value={cleaner.toLowerCase()} key={cleaner.toLowerCase()}>{cleaner}</option>
                        })
                     }
                  </select>
               </div>
               <div class="input-group mb-2">
                  <div class="input-group-prepend">
                     <label class="input-group-text" for="place">Place</label>
                  </div>
                  <select 
                     class="custom-select" 
                     id="place"
                     name="place" 
                     onChange={this.onPlaceChange}
                     value={this.state.place.toLowerCase()}
                  >
                     {
                        this.props.settings.places.map( place => {
                           return <option value={place.toLowerCase()} key={place}>{place}</option>
                        })
                     }
                  </select>
               </div>
               <div class="input-group mb-2">
                  <div class="input-group-prepend">
                     <label class="input-group-text" for="date">Date</label>
                  </div>
                  <SingleDatePicker 
                     date={this.state.date}
                     onDateChange={this.onDateChange}
                     focused={this.state.focused}
                     onFocusChange={this.onFocusChange}
                     numberOfMonths={1}
                     isOutsideRange={() => false}
                  />
               </div>
               <div class="form-group">
                  <textarea 
                     class="form-control"
                     value={this.state.note}
                     name="note" 
                     placeholder="Add your note"
                     onChange={this.onNoteChange}
                  ></textarea>
               </div>
               <div class="form-group text-right">
                  <Link class="text-danger mr-2" to={'/'}>Cancel</Link>
                  <button class="btn btn-primary">Submit</button>
               </div>
            </form>
         </div>
      )
   }
}

const mapStateToProps = ({ settings }) => ({
   settings: settings
})

export default connect(mapStateToProps)(AddEventForm)