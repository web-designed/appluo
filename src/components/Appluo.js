import React from 'react'
import EventsList from './EventsList'

class Appluo extends React.Component {
   render() {
      return (
         <div>
            <EventsList />
         </div>
      )
   }
}

export default Appluo