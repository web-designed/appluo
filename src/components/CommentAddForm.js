import React from 'react'
import { connect } from 'react-redux'
import uuid from 'uuid'
import { addComment } from '../actions/events'
import moment from 'moment'

export class CommentAddForm extends React.Component {

   state = {
      error: '',
      commentBody: '',
      commenter: 'kylu'
   }

   onCommentBodyChange = (e) => {
      const commentBody = e.target.value
      this.setState(() => ({ commentBody }))
   }

   handleSubmitComment = (e) => {

      e.preventDefault()

      if(!this.state.commentBody){
         this.setState(() => ({
            error: this.props.notifications.validate
         }))
      } else {
         const newComment = {
            commentBody: this.state.commentBody,
            commenter: this.state.commenter
         }
         //dispatch the action
         this.props.addComment(this.props.currentEventId, newComment)

         //clear the comment and the validation error
         this.setState(() => ({
            commentBody: '',
            error: ''
         }))
      }
   }

   render(){
      return (
         <div class="pt-3">
            <form onSubmit={this.handleSubmitComment}>
               <div class="form-group">
                  <textarea 
                     class="form-control bg-light"
                     name="commentBody"
                     value={this.state.commentBody}
                     onChange={this.onCommentBodyChange}
                     placeholder="Add your comment"
                  ></textarea>
                  {this.state.error && <p class="text-danger"><small>{this.state.error}</small></p>}
               </div>
               <div class="form-group">
                  <button class="btn btn-secondary">Add Comment</button>
               </div>
            </form>
         </div>
      )
   }
}

const mapDispatchToProps = dispatch => ({
   addComment: (id, newComment) => dispatch(addComment(id, newComment))
})

const mapStateToProps = state => ({
   notifications: state.settings.notifications.comment
})

export default connect(mapStateToProps, mapDispatchToProps)(CommentAddForm)