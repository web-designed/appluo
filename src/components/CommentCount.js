import React from 'react'

const CommentCount = ({ commentsCount }) => {
   return (
      <span>comments: <span>{ commentsCount }</span></span>
   )
}

export default CommentCount