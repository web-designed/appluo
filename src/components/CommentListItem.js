import React from 'react'
import moment from 'moment'

const CommentListItem = ({ comment }) => {

   if(comment){
      const now = moment()
      const fromNow = moment(comment.createdAt).from(now)

      return (
         <div class="list-group-item">
            <p>{comment.commentBody}</p>
            <div class="text-right">
               <small class="text-secondary">{comment.commenter} {fromNow}</small>
            </div>
         </div>
      )
   }
}

export default CommentListItem