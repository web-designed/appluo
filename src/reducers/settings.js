import moment from 'moment'

//*******************************************************
// SETTINGS REDUCER
//*******************************************************

   const settingsReducerDefaultState = {
      wgName: 'The Cleaning Team',
      slogan: 'Manage your cleaning plan',
      places: ['Kitchen', 'Bathroom', 'Saloon'],
      cleaners: ['John', 'Alex', 'Tom', 'Steve'],
      createdAt: moment('2018-12-01'),
      notifications: {
         event: {
            afterDelete: 'The cleaning was successfully removed',
            notFound: "The cleaning doesn't exist or has been removed"
         },
         comment: {
            validate: 'Please add your comment'
         }
      }
   }
   const settingsReducer = (state = settingsReducerDefaultState, action) => {
      switch(action.type){
         default:
            return state
      }
   } 

   export default settingsReducer