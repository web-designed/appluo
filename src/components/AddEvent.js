import React from 'react'
import AddEventForm from './AddEventForm'

const AddEvent = () => {
   return(
      <div>
         <h1>Add a cleaning event</h1>
         <AddEventForm />
      </div>
   )
}

export default AddEvent