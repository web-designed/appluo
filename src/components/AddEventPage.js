import React from 'react'
import AddEventForm from './AddEventForm'
import { addEvent } from '../actions/events'
import { connect } from 'react-redux'

const AddEventPage = (props) => (
   <div>
      <h1>Add a cleaning</h1>
      <AddEventForm 
         handleSubmit={ (event) => {
            props.onSubmit(event)
            props.history.push('/')
         }} 
         history={props.history} 
      />
   </div>
)

const mapDispatchToProps = (dispatch) => ({
   onSubmit: (expense) => dispatch(addEvent(expense))
})

export default connect(undefined, mapDispatchToProps)(AddEventPage)