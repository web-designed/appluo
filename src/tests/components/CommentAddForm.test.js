import React from 'react'
import { shallow } from 'enzyme'
import { CommentAddForm } from '../../components/CommentAddForm'
import settings from '../fixtures/settings'

let wrapper, addComment, form, currentEventId, value

beforeEach(() => {
   currentEventId = '1'
   addComment = jest.fn()
   wrapper = shallow(<CommentAddForm currentEventId={currentEventId} addComment={addComment} notifications={settings.notifications.comment} />)
   form = wrapper.find('form')
   value = 'my new comment'
})

test('should render the CommentAddForm correctly', () => {
   expect(wrapper).toMatchSnapshot()
})

test('should set the validation message if a not valid form submited', () => {
   form.simulate('submit', {
      preventDefault: () => {}
   })
   expect(wrapper.state('error')).toBe(settings.notifications.comment.validate)
})

test('should set commentBody state on onCommentBodyChange', () => {
   form.find('textarea').simulate('change', {
      target: { value }
   })
   expect(wrapper.state('commentBody')).toBe(value)
})

test('should handle handleSubmitComment', () => {
   form.find('textarea').simulate('change', {
      target: { value }
   })
   form.simulate('submit', {
      preventDefault: () => {}
   })
   const newComment = {
      commentBody: value,
      commenter: 'kylu'
   }
   expect(addComment).toHaveBeenLastCalledWith('1', newComment)
   expect(wrapper.state('error')).toBe('')
   expect(wrapper.state('commentBody')).toBe('')
})