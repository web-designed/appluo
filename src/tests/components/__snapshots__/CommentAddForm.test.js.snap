// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the CommentAddForm correctly 1`] = `
<div
  class="pt-3"
>
  <form
    onSubmit={[Function]}
  >
    <div
      class="form-group"
    >
      <textarea
        class="form-control bg-light"
        name="commentBody"
        onChange={[Function]}
        placeholder="Add your comment"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <button
        class="btn btn-secondary"
      >
        Add Comment
      </button>
    </div>
  </form>
</div>
`;
